lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1533,26 +1522,36 @@ export class UserAgentApplication {
       }
       else {
         // if more than cached token is found
-        // TODO: Return custom error here
-        // TODO: Check that accessToken is only possible tokenType for this error type
-        return {
-          errorDesc: "The cache contains multiple tokens satisfying the requirements.Call AcquireToken again providing more requirements like authority",
-          token: null,
-          error: "multiple_matching_tokens_detected"
-        };
+        throw ClientAuthError.createMultipleMatchingTokensInCacheError(scopes.toString());
       }
     }
 
     if (accessTokenCacheItem != null) {
-      const expired = Number(accessTokenCacheItem.value.expiresIn);
+      let expired = Number(accessTokenCacheItem.value.expiresIn);
       // If expiration is within offset, it will force renew
       const offset = this.config.system.tokenRenewalOffsetSeconds || 300;
       if (expired && (expired > Utils.now() + offset)) {
-        return {
-          errorDesc: null,
-          token: accessTokenCacheItem.value.accessToken,
-          error: null
+        const idToken = new IdToken(accessTokenCacheItem.value.idToken);
+        if (!account) {
+          account = this.getAccount();
+          if (!account) {
+            throw AuthError.createUnexpectedError("Account should not be null here.");
+          }
+        }
+        const aState = this.getAccountState(this.cacheStorage.getItem(Constants.stateLogin, this.inCookie));
+        let response : AuthResponse = {
```

Comment:
can we pls discuss this?

Reply:
Any open questions?

Reply:
we should standardize the storage of the expiry in the cache such that it's consistent, something like seconds since epoch. looks like currently, it's either epoch time in case of id_token, and duration in case of access_token. With that, please make sure every caller/usage of this is modified appropriately.

Reply:
After reviewing this further, Utils.now() returns time in Unix time (seconds), which is the same as the format of the exp claim in the id token.

Reply:
Therefore, as long as we use utils.now when we are storing the access tokens, we can always be sure that the cache is storing expiration time in seconds.